src/Akka.Streams.Kafka.Tests/Integration/ExternalPlainSourceIntegrationTests.cs

```diff
@@ -0,0 +1,159 @@
+            await ProduceStrings(new TopicPartition(topic, new Partition(1)), Enumerable.Range(1, elementsCount), ProducerSettings);
+
+            // Request for produced messages and consume them
+            probe1.Request(elementsCount);
+            probe2.Request(elementsCount);
+            probe1.Within(TimeSpan.FromSeconds(10), () => probe1.ExpectNextN(elementsCount));
+            probe2.Within(TimeSpan.FromSeconds(10), () => probe2.ExpectNextN(elementsCount));
+           
+            // Stop stages
+            probe1.Cancel();
+            probe2.Cancel();
+            
+            // Make sure stages are stopped gracefully
+            AwaitCondition(() => partitionTask1.IsCompletedSuccessfully && partitionTask2.IsCompletedSuccessfully);
+            
+            // Cleanup
+            consumer.Tell(new KafkaConsumerActorMetadata.Internal.Stop(), ActorRefs.NoSender);
+        }
+
+        [Fact]
+        public async Task ExternalPlainSource_should_be_stopped_on_serialization_error_only_when_requested_messages()
+        {
+            var topic = CreateTopic(1);
+            var group = CreateGroup(1);
+
+            // Make consumer expect numeric messages
+            var settings = CreateConsumerSettings<int>(group).WithValueDeserializer(Deserializers.Int32);
+            var consumer = Sys.ActorOf(KafkaConsumerActorMetadata.GetProps(settings));
+            
+            // Subscribe to partitions
+            var (partitionTask1, probe1) = CreateProbe<int>(consumer, Subscriptions.Assignment(new TopicPartition(topic, 0)));
+            var (partitionTask2, probe2) = CreateProbe<int>(consumer, Subscriptions.Assignment(new TopicPartition(topic, 1)));
+            var (partitionTask3, probe3) = CreateProbe<int>(consumer, Subscriptions.Assignment(new TopicPartition(topic, 2)));
+
+            // request from 2 streams
+            probe1.Request(1);
+            probe2.Request(1);
+            await Task.Delay(500); // To establish demand
+
+            // Send string messages
+            await ProduceStrings(new TopicPartition(topic, 0), new int[] { 1 }, ProducerSettings);
+            await ProduceStrings(new TopicPartition(topic, 1), new int[] { 1 }, ProducerSettings);
+
+            // First two stages should fail, and only stage without demand should keep going
+            probe1.ExpectError().Should().BeOfType<SerializationException>();
+            probe2.ExpectError().Should().BeOfType<SerializationException>();
+            probe3.Cancel();
+            
+            // Make sure source tasks finish accordingly
+            AwaitCondition(() => partitionTask1.IsFaulted && partitionTask2.IsFaulted && partitionTask3.IsCompletedSuccessfully);
+            
+            // Cleanup
+            consumer.Tell(new KafkaConsumerActorMetadata.Internal.Stop(), ActorRefs.NoSender);
+        }
+
+        [Fact]
+        public async Task ExternalPlainSource_verify_consuming_actor_pause_resume_partitions_works_fine()
```

Comment:
This test is specially implemented for issue #55 - there is no corresponding test in alpakka project.

Reply:
Great, thank you